File: frontend/src/services/goalService.js
import apiClient from './apiClient.js';

export async function getGoals(status) {
  const { data } = await apiClient.get('/goals', { params: status ? { status } : undefined });
  return data.data;
}

export async function createGoal(payload) {
  const { data } = await apiClient.post('/goals', payload);
  return data.data;
}

export async function updateGoal(id, payload) {
  const { data } = await apiClient.put(`/goals/${id}`, payload);
  return data.data;
}

export async function deleteGoal(id) {
  await apiClient.delete(`/goals/${id}`);
}

export async function addContribution(id, payload) {
  const { data } = await apiClient.post(`/goals/${id}/contributions`, payload);
  return data.data;
}

export async function getContributions(id) {
  const { data } = await apiClient.get(`/goals/${id}/contributions`);
  return data.data;
}

File: backend/src/main/java/com/smartmoneymanager/backend/service/GoalService.java
package com.smartmoneymanager.backend.service;

import java.util.List;

import com.smartmoneymanager.backend.dto.request.AddGoalContributionRequest;
import com.smartmoneymanager.backend.dto.request.CreateGoalRequest;
import com.smartmoneymanager.backend.dto.request.UpdateGoalRequest;
import com.smartmoneymanager.backend.dto.response.GoalContributionResponse;
import com.smartmoneymanager.backend.dto.response.GoalResponse;
import com.smartmoneymanager.backend.entity.enums.GoalStatus;

public interface GoalService {

    List<GoalResponse> getGoals(Long userId, GoalStatus status);

    GoalResponse getGoal(Long userId, Long goalId);

    GoalResponse createGoal(Long userId, CreateGoalRequest request);

    GoalResponse updateGoal(Long userId, Long goalId, UpdateGoalRequest request);

    void deleteGoal(Long userId, Long goalId);

    GoalResponse addContribution(Long userId, Long goalId, AddGoalContributionRequest request);

    List<GoalContributionResponse> getContributions(Long userId, Long goalId);
}
